src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/processors/PrisonerUpdatedEventProcessor.kt

```diff
@@ -0,0 +1,18 @@
+package uk.gov.justice.digital.hmpps.personrecord.message.listeners.processors
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.personrecord.model.DomainEvent
+
+const val PRISONER_UPDATED = "probation-case.engagement.created"
+
+@Component(value = PRISONER_UPDATED)
+class PrisonerUpdatedEventProcessor() : EventProcessor() {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+  override fun processEvent(domainEvent: DomainEvent) {
+    val nomsNumber = domainEvent.additionalInformation?.nomsNumber!!
+    log.debug("Entered processEvent with  $nomsNumber")
+  }
+}
```

Comment:
Given what we discussed yesterday, I don't think we need separate processors for these two events as we are going to treat them in the same way (apart from logging)

Reply:
```
 if(domainEvent.eventType.equals("probation-case.engagement.created") {
      return context.getBean(domainEvent.eventType)
    }
    return PrisonerCreatedEventProcessor() // or whatever we call it.
```

This should do the trick without changing the design too much - we are just introducing complexity we will not need (unless we change course again :-)) by having 2 processors

Reply:
Those two are the beans for each type. Isn't? we can have a chat.

Reply:
In my opinion, each event type should be a bean which can still share the same service/ implementation etc..